### PR TITLE
Ensure fontList is set for a previously embedded font

### LINKED ISSUE #902: fontName in style not working with fontsAlias
It seems that specifying a font as `stdFont` in `fontsAlias`, then later using `fontName: stdFont` in some style doesn't work.
Instead of using the alias it seems to be taking `stdFont` as a literal font name.

**Description of problem**

1. Download the rst and style files for "Document with Simple Styling" from the [rst2pdf examples page](https://rst2pdf.org/examples).
2. Generate the pdf
3. Change the `fontName` for the `normal` style to be literally `Georgia`
4. Regenerate the pdf
5. Notice that the font changed, even though `stdFont` should've been pointing to the `Georgia` font.

**What is the expected output? What do you see instead?**
Using `stdFont`, which is `Georgia`, or using directly `Georgia` should both yield the same pdf.

🖥 **Versions**

`python -V`: Python 3.8.5

`pip freeze | grep rst2pdf`: rst2pdf==0.97.dev0

`pip freeze | grep reportlab`: reportlab==3.5.48

**Which operating system are you using?**
Archlinux 5.8.3-arch1-1

**Additional information**

I don't actually have the Georgia font, but using `pdffonts document1-simple.pdf` (at step 2.) I can see that the font used was `DejaVuSans`, even though `fc-match Georgia` yields `DejaVu Serif`.
Using `pdffonts document1-simple.pdf` again at step 4. shows `DejaVuSerif`.
Also, `fc-match stdFont` returns `DejaVu Sans`, so that's why I think `stdFont` is being taken literally as the font name.

### PR DESCRIPTION
**Pull request by akrabat**
If a font is aliased and then that font name is subsequently used as a style name, then the `fontList` variable will not be set which will result in the font name becoming an alias to whichever font happened to be in `fontList` the previous time around the loop.

Example style sheet:

```
fontsAlias:
    stdFont: inconsolata
    stdBold: carlito

styles:
    normal:
        fontName: inconsolata
        textColor: #383745
        fontSize: 16
        alignment: TA_LEFT

    heading:
        fontName: stdFont
        fontSize: 32
        textColor: #1a8099
```

In this example, `inconsolata` is set as an alias to `stdFont` which means that `inconsolata` is registed as the key name for the embedded font list pointing to the correct font file names for that font (`Inconsolata-Regular_9ba2497f55c3435c544f6702d57e7a94` on my machine). However, `inconsolata` is also referenced as the `fontName` for the `normal` style which means that without this fix, the embedded font referenced by the `inconsolata` key now points the last font that was processed (`Cardo` in this case).

Fixes #902.

Note: The CI will fail until #909 is merged (and ideally #910 is fixed). I will then rebase.

### CODE REVIEW

**Comment by nfraprado** [comment]
I tested the change in `rst2pdf/styles.py` and it works as expected, fixing the bug.

**Comment by akrabat** [comment]
Rebased against master now that #909 has been merged.

File: rst2pdf/styles.py [hunk 1]
```diff
@@ -363,7 +363,7 @@ def __init__(self, flist, font_path=None, style_path=None, def_dpi=300):
                         fname, pos = findfonts.guessFont(style[key])
 
                         if style[key] in embedded_fontnames:
-                            pass
+                            fontList = findfonts.autoEmbed(style[key])
                         else:
                             fontList = findfonts.autoEmbed(style[key])
```
Comment (stephenfin) [inline rst2pdf/styles.py:368]: Seeing as these are now identical, we could move them out of the conditional are simply retain the rest of the else block.

```
fontList = findfonts.autoEmbed(style[key])
if style[key] not in embedded_fontnames:
    if fontlist:
        # ...
```

**Review by stephenfin — APPROVED** [review]
This looks sensible. One comment inline but nothing I'd hold this up over

**Review by stephenfin — APPROVED** [review]
Still good

**Review by lornajane — APPROVED** [review]
Yep, works for me. Thanks for fixing :+1:

**Pull request merged**